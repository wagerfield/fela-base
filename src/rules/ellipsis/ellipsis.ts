import { getRuleProps } from "../defaults"

export interface EllipsisRuleProps {
  textOverflow?: string
  whiteSpace?: string
  overflow?: string
}

export const NAME = "ellipsis"
export const DEFAULTS = {
  textOverflow: "ellipsis",
  whiteSpace: "nowrap",
  overflow: "hidden"
}

export default (props?: EllipsisRuleProps) => {
  const p = getRuleProps(NAME, DEFAULTS, props)
  return {
    textOverflow: p.textOverflow,
    whiteSpace: p.whiteSpace,
    overflow: p.overflow
  }
}
