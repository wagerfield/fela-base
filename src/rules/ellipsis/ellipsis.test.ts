import ellipsis from "./ellipsis"

describe("ellipsis(props)", () => {
  it("renders expected snapshots", () => {
    expect(ellipsis()).toMatchSnapshot()
  })
})
