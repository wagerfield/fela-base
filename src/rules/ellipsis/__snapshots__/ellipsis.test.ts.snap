// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ellipsis(props) renders expected snapshots 1`] = `
Object {
  "overflow": "hidden",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;
